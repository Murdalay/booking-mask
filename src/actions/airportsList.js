// @flow
/**
 * @module Actions/airportsList
 * @desc airportsList Actions
 */
import { createActions } from 'redux-actions';

import { ActionTypes } from 'constants/index';

export const { airportsListGet } = createActions({
  [ActionTypes.AIRPORTS_LIST_GET]: () => ({}),
});
