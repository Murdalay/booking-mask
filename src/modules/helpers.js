// @flow
/**
 * Helper functions
 * @module Helpers
 */

/**
 * Convert data attributes to Object
 * @param {Element} elem
 * @returns {{}}
 */
export function datasetToObject(elem: Element): Object {
  const data = {};
  [].forEach.call(elem.attributes, attr => {
    /* istanbul ignore else */
    if (/^data-/.test(attr.name)) {
      const camelCaseName = attr.name.substr(5).replace(/-(.)/g, ($0, $1) => $1.toUpperCase());
      data[camelCaseName] = attr.value;
    }
  });
  return data;
}

/**
 * Construct the url basing on booking object and prefix
 * @param {Object} booking â€“ `booking` state field object
 * @param {String} prefix - URL base
 * @param {String} [postFix] - URL ending
 * @returns {String}
 */
export function constructUrlFromBooking(
  booking: Object,
  prefix: string,
  postFix: string = '',
): string {
  const passangers = `adults-${booking.passangers.adults}/children-${
    booking.passangers.children
  }/infants-${booking.passangers.infants}`;
  const outboundFlight = `${booking.from}-${booking.to}/from-${booking.flightDates.departure}`;
  const returnFlight =
    booking.isRoundTrip && booking.flightDates.return
      ? `/Return/${booking.to}-${booking.from}/from-${booking.flightDates.return}`
      : '';

  return `${prefix}/Outbound/${outboundFlight}${returnFlight}/${passangers}/${postFix}`;
}
