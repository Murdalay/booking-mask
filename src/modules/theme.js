export const rowHeight = 70;
export const controlHeight = 55;
export const appColor = '#00b4d5';
export const secondaryColor = '#88e4f5';
export const actionColor = '#66b4e5';
export const inputMinWidth = 150;
export const defaultIconColor = '#444';
export const defaultTextColor = '#555';
export const borderColor = 'rgb(204, 204, 204)';
export const easing = 'cubic-bezier(0.35, 0.01, 0.77, 0.34);';
export const inputFontSize = '1.5rem';
export const inputBackgroundColor = '#fff';

export default {
  breakpoints: [400, 768, 1024, 1280, 1920],
  palette: {
    primary: appColor,
    secondary: secondaryColor,
    actionColor,
    appColor,
    borderColor,
    defaultTextColor,
    defaultIconColor,
    inputBackgroundColor,
  },
  sizes: {
    rowHeight,
    controlHeight,
    borderColor,
    inputMinWidth,
    inputFontSize,
  },
  input: {
    borderRadius: {
      xs: 0,
      lg: 0,
      xl: 0,
      default: 0,
    },
  },
  button: {
    borderRadius: {
      xs: 0,
      lg: 0,
      xl: 0,
    },
    padding: {
      lg: [12, 28],
      xl: [14, 32],
    },
  },
};
