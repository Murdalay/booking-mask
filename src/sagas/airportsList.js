/**
 * @module Sagas/airportsList
 * @desc airportsList
 */

import { all, call, put, takeLatest } from 'redux-saga/effects';
import { mimicRequest } from 'modules/client';
import { ActionTypes } from 'constants/index';

/**
 * Get Repos
 */
export function* getAiroportsList() {
  try {
    const response = yield call(mimicRequest, 'https//sample-url');
    yield put({
      type: ActionTypes.AIRPORTS_LIST_GET_SUCCESS,
      payload: { data: response },
    });
  } catch (err) {
    /* istanbul ignore next */
    yield put({
      type: ActionTypes.AIRPORTS_LIST_GET_FAILURE,
      payload: err,
    });
  }
}

/**
 * airportsList Sagas
 */
export default function* root() {
  yield all([takeLatest(ActionTypes.AIRPORTS_LIST_GET, getAiroportsList)]);
}
