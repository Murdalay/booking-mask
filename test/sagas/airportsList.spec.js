import { expectSaga } from 'redux-saga-test-plan';
import airportsList, { getAiroportsList } from 'sagas/airportsList';
import { mimicRequest } from 'modules/client';

describe('airportsList', () => {
  it('should have the expected watchers', done =>
    expectSaga(airportsList)
      .run({ silenceTimeout: true })
      .then(saga => {
        expect(saga).toMatchSnapshot();
        done();
      }));

  it('get airoports list saga should call mimicRequest with right params', () =>
    expectSaga(getAiroportsList, { payload: {} })
      .provide({
        call(effect, next) {
          // Check for the API call to return fake value
          if (effect.fn === mimicRequest) {
            expect(effect.args[0]).toBe('https//sample-url');
          }

          // Allow Redux Saga to handle other `call` effects
          return next();
        },
      })

      .run());

  it('get airoports list saga should put AIRPORTS_LIST_GET_SUCCESS', () => {
    jest.mock('modules/client', () => ({
      mimicRequest: () => ['foo', 'bar', 'baz'],
    }));

    expectSaga(getAiroportsList, { payload: {} })
      .put({
        type: 'AIRPORTS_LIST_GET_SUCCESS',
        payload: { data: ['foo', 'bar', 'baz'] },
      })
      .run();
  });

  it('get airoports list saga should put AIRPORTS_LIST_GET_FAILURE on error', () => {
    jest.mock('modules/client', () => ({
      mimicRequest: () => {
        throw new Error('ups...');
      },
    }));

    expectSaga(getAiroportsList, { payload: {} })
      .put({
        type: 'AIRPORTS_LIST_GET_FAILURE',
        payload: {},
      })
      .run();
  });
});
