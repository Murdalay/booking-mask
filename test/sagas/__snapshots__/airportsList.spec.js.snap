// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`airportsList should have the expected watchers 1`] = `
Object {
  "fork": Array [
    Object {
      "@@redux-saga/IO": true,
      "FORK": Object {
        "args": Array [
          "AIRPORTS_LIST_GET",
          "@@redux-saga-test-plan/json/function/getAiroportsList",
        ],
        "context": null,
        "fn": "@@redux-saga-test-plan/json/function/takeLatest",
      },
    },
  ],
  "take": Array [
    Object {
      "@@redux-saga/IO": true,
      "TAKE": Object {
        "pattern": "AIRPORTS_LIST_GET",
      },
    },
  ],
}
`;
